components/SubmissionComments.test.js

```diff
@@ -105,4 +139,82 @@ describe('Test SubmissionComments Component', () => {
       expect(deleteCommentMutation).toHaveBeenCalled()
     })
   })
+
+  it('should allow user to edit comment and discard changes when edit button is pressed, but not edit more than 1 comment in a chain', async () => {
+    const cache = new InMemoryCache({ addTypename: false })
+
+    cache.writeQuery({
+      query: GET_PREVIOUS_SUBMISSIONS,
+      variables: { userId: 3, challengeId: 6 },
+      data: { getPreviousSubmissions: submissionsData }
+    })
+    expect.assertions(3)
+    const { container } = render(
+      <MockedProvider cache={cache} addTypename={false} mocks={mocks}>
+        <Wrapper>
+          <SubmissionComments comments={comments} submission={submissionData} />
+        </Wrapper>
+      </MockedProvider>
+    )
+
+    const editButton = container.querySelector('.btn-info.btn-sm')
+    fireEvent.click(editButton)
+
+    await waitFor(() => {
+      const discardButton = container.querySelector('.btn-light')
+      expect(discardButton).toBeTruthy()
+      expect(container.querySelector('.btn-info')).toBeTruthy()
+
+      const editButtonTwo = container.querySelector('.btn-info.btn-sm')
+      fireEvent.click(editButtonTwo)
+
+      waitFor(() => {
+        expect(Alert.alert).toHaveBeenCalledWith(
+          'You can only edit one comment in a single comment chain at a time.'
+        )
+      })
+
+      fireEvent.click(discardButton)
+      waitFor(() => {
+        expect(editButton).toBeTruthy()
+      })
+    })
+  })
+  it('should update comment when save changes button is clicked', async () => {
+    const cache = new InMemoryCache({ addTypename: false })
+
+    cache.writeQuery({
+      query: GET_PREVIOUS_SUBMISSIONS,
+      variables: { userId: 3, challengeId: 6 },
+      data: { getPreviousSubmissions: submissionsData }
+    })
+    const { container, getByTestId } = render(
+      <MockedProvider cache={cache} addTypename={false} mocks={mocks}>
+        <Wrapper>
+          <SubmissionComments
+            comments={comments.slice(0, 1)}
+            submission={submissionData}
+          />
+        </Wrapper>
+      </MockedProvider>
+    )
+
+    const editButton = container.querySelector('.btn-info.btn-sm')
```

Comment:
In the future, we should update the `Button` component to take a `data-testid` prop so we can use `getByTestId('edit-button')`.